internal/util/setters/setters.go

```diff
@@ -133,58 +150,54 @@ func SetV1AutoSetter(name, value, path string) error {
 
 // SetV2AutoSetter sets the input auto setter recursively in all the sub-packages of root
 // Sets GcloudProjectNumber as well, if input setter is GcloudProject
-func SetV2AutoSetter(name, value, root string) error {
+func SetV2AutoSetter(name, value, root string, w io.Writer) error {
 	resourcePackagesPaths, err := pathutil.DirsWithFile(root, kptfile.KptFileName, true)
 	if err != nil {
 		return err
 	}
 	for _, resourcesPath := range resourcePackagesPaths {
-		fs := &settersutil.FieldSetter{
-			Name:               name,
-			Value:              value,
-			SetBy:              "kpt",
-			OpenAPIPath:        filepath.Join(resourcesPath, kptfile.KptFileName),
-			OpenAPIFileName:    kptfile.KptFileName,
-			ResourcesPath:      resourcesPath,
-			RecurseSubPackages: true,
+		if isSet(name, filepath.Join(resourcesPath, kptfile.KptFileName)) {
```

Comment:
Is it safe to rely on the `isSet` property here? I'm both thinking if there are situations where to user might have updated the value, but it should still be covered by the autosetter, and if there are situations where an upstream package might already have the `isSet` property set.

Reply:
`isSet` is only checked for the local package which is being set. So the second point you mentioned is taken care of.

I don't see any cases where we should set the value after the value has been set locally.